modules/auth-on-llama-stack.adoc

```diff
@@ -0,0 +1,324 @@
+metadata:
+  name: llamastack-distribution
+  namespace: redhat-ods-operator
+spec:
+  replicas: 1
+  server:
+    distribution:
+      name: rh-dev
+    containerSpec:
+      name: llama-stack
+      port: 8321
+      env: 
+        # vLLM Provider Configuration
+        - name: VLLM_URL
+          value: "http://your-vllm-service:8000/v1"
+        - name: VLLM_API_TOKEN
+          value: "your-vllm-token"
+        - name: VLLM_TLS_VERIFY
+          value: "false"
+        # OpenAI Provider Configuration
+        - name: OPENAI_API_KEY
+          value: "your-openai-api-key"
+        - name: OPENAI_BASE_URL
+          value: "https://api.openai.com/v1"
+    userConfig:
+      configMapName: llamastack-custom-config
+      configMapNamespace: redhat-ods-operator
+----
+
+. To apply the distribution, run the following command: 
++
+[source,terminal]
+----
+$ oc apply -f llamastack-distribution.yaml
+----
+
+. Wait for the distribution to be ready by running the following command: 
++
+[source,terminal]
+----
+oc wait --for=jsonpath='{.status.phase}'=Ready llamastackdistribution/llamastack-distribution -n redhat-ods-operator --timeout=300s
+----
+
+. Generate the OAuth tokens for each service account to authenticate API requests. 
+
+.. To request a basic access token and add it to a `user1.token` file, run the following command
++
+[source,terminal]
+----
+$ curl -d client_id=llamastack \
+  -d client_secret=YOUR_CLIENT_SECRET \
+  -d username=user1 \
+  -d password=user1-password \
+  -d grant_type=password \
+  https://YOUR_KEYCLOAK_HOST/realms/YOUR_REALM/protocol/openid-connect/token \
+  | jq -r .access_token > user1.token
+----
+
+.. To request full access token and add it to a `user1.token` file, run the following command
++
+[source,terminal]
+----
+$ curl -d client_id=llamastack \
+  -d client_secret=YOUR_CLIENT_SECRET \
+  -d username=user2 \
+  -d password=user2-password \
+  -d grant_type=password \
+  https://YOUR_KEYCLOAK_HOST/realms/YOUR_REALM/protocol/openid-connect/token \
+  | jq -r .access_token > user2.token
+----
+
+.. You can verify the credentials by running the following command: 
++
+[source,terminal]
+----
+$ cat user2.token | cut -d . -f 2 | base64 -d 2>/dev/null | jq .
+----
++
+.Example output 
+[source,terminal]
+----
+{
+  "iss": "https://keycloak-host/realms/testrealm",
+  "aud": "account",
+  "exp": 1760553504,
+  "preferred_username": "user2",
+  "llamastack_roles": ["inference_max"]
+}
+----
+
+.Verification 
+
+* Testing basic au to models
+
+. Load the token with the following command: 
++
+[source,terminal]
+----
+USER1_TOKEN=$(cat user1.token)
+----
+
+. Access the vLLM model by running the following command: 
++
+[source,terminal]
+----
+curl -X POST ${LLAMASTACK_URL}/v1/openai/v1/chat/completions -H "Content-Type: application/json" -H "Authorization: Bearer ${USER1_TOKEN}" -d '{"model": "vllm-inference/llama-3.2", "messages": [{"role": "user", "content": "Hello!"}], "max_tokens": 50}'
```

Comment:
We need to set $LLAMASTACK_URL somewhere and to do that we need to create a route.

Reply:
Is there a spot/line you think would be good to add this from procedureal perspective? @derekhiggins